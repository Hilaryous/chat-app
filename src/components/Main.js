// @flow
import React, { Fragment } from 'react'
import styled from 'react-emotion'
import MessageForm from './MessageForm'
import MessageList from './MessageList'
import ChatList from './ChatList'
import Nav from './Nav'

const Main = () => (
  (
    <Fragment>
      <Nav />
      <Container>
        <div>
          <MessageList />
          <MessageForm />
        </div>
        <ChatList />
      </Container>
    </Fragment>
  )
)


export default Main

// --------------------------

const Container = styled('div')({
  display: 'grid',
  gridTemplateColumns: '1fr',
  '@media only screen and (min-width: 35em)': {
    gridTemplateColumns: '3fr 1fr',
  },
})
