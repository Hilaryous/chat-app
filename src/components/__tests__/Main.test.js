import React from 'react'
import renderer from 'react-test-renderer'
import { StaticRouter } from 'react-router-dom'
import Main from '../Main'
import AppContext from '../../context'

describe('Main', () => {
  it('shows the correct snapshot tree for the Main component', () => {
    const tree = renderer.create(
      <StaticRouter location="someLocation" context={{}}>

        <AppContext.Provider>
          <Main />
        </AppContext.Provider>
      </StaticRouter>,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
