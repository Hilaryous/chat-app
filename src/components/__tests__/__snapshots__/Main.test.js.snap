// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main shows the correct snapshot tree for the Main component 1`] = `
Array [
  <div
    className="css-1b2b698"
  >
    <div
      className="css-1bbbaa"
      height="100%"
    >
      <p>
        Hi! 
      </p>
      <button
        className="css-bjolns"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>,
  <div
    className="css-1ylfck5"
  >
    <div>
      <div />
    </div>
    <div />
  </div>,
]
`;
